### DISCUSSION TITLE
Custom output library name

### DISCUSSION THREAD

**Discussion by PierBJX** [webpack/webpack#14923] [Q&A] [answered]
I have a webpack common configuration which takes some parameters as input. And I would like to build output library according to the entry name. However, I would like to convert the entry name which is in `kebab-case` to `camelCase` for `root`. And I don't know how to do it with [name].

Here is the common webpack config:

```
export default (options) => {
  const { entry, scope } = options;

  return {
    entry,
    output: output({
        filename: '[name].js',
        library: {
          commonjs: `@${scope}/[name]`,
          amd: `@${scope}/[name]`,
          root: [scope.replace(/-[a-z]/g, m => m[1].toUpperCase()), '[name]'.replace(/-[a-z]/g, m => m[1].toUpperCase())],
        },
      },
    ),
   ...
  };
};
```

**Comment by alexander-akait** [comment, marked as answer]
You can't do it, `name` is the same as your provide

**Reply by PierBJX** [reply]
But how `name` is interpreted ? It is not a string? For example, in the ```commonjs: `@${scope}/[name]`,```

**Reply by alexander-akait** [reply]
it is placholder, which will be replaced on the name
